pkgs/servers/sql/postgresql/default.nix

```diff
@@ -22,14 +22,18 @@ let
 
     makeFlags = [ "world" ];
 
+    preConfigure = ''
+    export NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE -I${libxml2.dev}/include/libxml2"
```

Comment:
Does postgresql also picks this up, when `pkgconfig` is included in `nativeBuildInputs`?

Reply:
Now I see why this fails: 

  https://github.com/postgres/postgres/blob/master/configure.in#L896

It needs `xml2-config` compiled for the build platform.

Reply:
Instead of preConfigure, this can be exported in nix instead:

```nix
# needed for cross-compiling, since xml2-config is not compiled for the build platform
NIX_CFLAGS_COMPILE = [ "-I${libxml2.dev}/include/libxml2" ];
```

Reply:
The long-term solution would be `pkg-config`: However this would been an upstream fix: https://github.com/libarchive/libarchive/issues/407

Reply:
I see. I will try to raise this issue in `postgresql` mailing list.

Reply:
Actually `xml2-config` is built correctly but it doesn't get into the `PATH` since we don't add binaries from `dev` output to `PATH`. This is something we should do in stdenv since many other packages will benefit from that as well.

Reply:
Should I do it in a separate pull request? Or this one?

Reply:
Yes I agree it can be done later.

Let's however conditionalize it on cross, so that someone doesn't break it by noticing it native-compiles fine with it.